### PR TITLE
PODS-469 Reg PSA (UK) - Company - contact details

### PR DESCRIPTION
**Pull request by madhura005**

### CODE REVIEW

File: conf/messages.en [hunk 1]
```diff
@@ -44,6 +44,18 @@ site.secondaryHeader = Register a new Pension Scheme Administrator
...
+contactDetails.email = Email address
+contactDetails.phone = Phone number
+contactDetails.body = We will use this information to contact the PSA about the schemes they manage.
+contactDetails.checkYourAnswersLabel = PSA contact details
+contactDetails.error.email.required = Enter an email address
+contactDetails.error.email.invalid = Enter a valid email address
+contactDetails.error.phone.required = Enter a phone number
+contactDetails.error.phone.length = The telephone number cannot be more than 24 characters
+contactDetails.error.phone.invalid = The telephone number can only contain numbers
+
```
Comment (PaulCDurham) [inline conf/messages.en]: Missing contactDetails.error.email.length - this is referenced from within the code

File: test/forms/mappings/ConstraintsSpec.scala [hunk 2]
```diff
@@ -156,4 +156,44 @@ class ConstraintsSpec extends WordSpec with MustMatchers with Constraints with R
...
   }
+
+  "email" must {
+
+    val validEmail = Table(
+      "a@email.com",
+      "a@bc",
+      "123@456"
+    )
+
```
Comment (PaulCDurham) [inline test/forms/mappings/ConstraintsSpec.scala]: First row in Table should be a title not a test value. So should be something like "email". Same for invalidEmail - change "value" to "email".

File: test/forms/mappings/ConstraintsSpec.scala [hunk 3]
```diff
@@ -156,4 +156,44 @@ class ConstraintsSpec extends WordSpec with MustMatchers with Constraints with R
...
+  }
+
+  "wholeNumber" must {
+
+    val validNumber = Table(
+      "1",
+      "99999999999999999999999",
+      "123456"
+    )
+
```
Comment (PaulCDurham) [inline test/forms/mappings/ConstraintsSpec.scala]: Same as validEmail Table. Add a first row that is the name of the data item, eg "number". Make invalidNumber the same.

File: test/views/register/company/ContactDetailsViewSpec.scala [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      controllers.register.company.routes.ContactDetailsController.onSubmit(NormalMode).url,
+      "email",
+      "phone"
+    )
+
+    "display correct body text" in {
+      assertContainsText(Jsoup.parse(createView().toString()), messages(s"$messageKeyPrefix.body"))
+    }
+  }
+}
```
Comment (PaulCDurham) [inline test/views/register/company/ContactDetailsViewSpec.scala]: There is no test that the view has a back button. I think this is the scaffold's fault. There is no back button on the view either. To match the prototype this should be added to both view and spec.

**Comment by oliverslade** [comment]
@PaulCDurham this should be ready to merge now

**Pull request merged**